lib/twiml/TwiML.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+var builder = require('xmlbuilder');  /* jshint ignore:line */
+
+/* jshint ignore:start */
+/**
+ * Parent TwiML object
+ */
+/* jshint ignore:end */
+function TwiML() {
+  this.response = builder.create('Response').dec('1.0', 'UTF-8');
+}
+
+/* jshint ignore:start */
+/**
+ * Because child elements have properties named after their class names, we need
+ * to translate that when we want to access that at the parent prototype level.
+ * So this will translate something like Say to 'say' and VoiceResponse to
+ * 'response'.
+ *
+ * It will also soft-cache it on the object so we don't end up doing string
+ * comparisons. Save a few CPU cycles.
+ */
+/* jshint ignore:end */
+TwiML.prototype._getPropertyName = function _getPropertyName() {
+  if (this._propertyName === undefined) {
+    var className = this.constructor.name.toLowerCase();
+
+    if (className.substr(-8) === 'response') {
+      className = 'response';
+    }
+
+    this._propertyName = className;
+  }
+
+  return this._propertyName;
+}
+
+/* jshint ignore:start */
+/**
+ * Convert to TwiML
+ *
+ * @returns TwiML XML
+ */
+/* jshint ignore:end */
+TwiML.prototype.toString = function toString() {
+  return this[this._getPropertyName()].end();
```

Comment:
Is there a way to have a non-public `getXml` that returns `this[this._getPropertyName()]` to avoid duplication?

Reply:
Duplication of what?

Reply:
the code `this[this._getPropertyName()]` appears in several places -- I thought it would be nice to have a helper method to avoid repeating it

Reply:
Ah, yes, great point!